interface/gemm.c

```diff
@@ -485,9 +492,38 @@ void CNAME(enum CBLAS_ORDER order, enum CBLAS_TRANSPOSE TransA, enum CBLAS_TRANS
   }
 #endif
 #endif  // defined(__linux__) && defined(__x86_64__) && defined(BFLOAT16)
+ //         fprintf(stderr,"G E M M interface m n k %d %d %d\n",args.m,args.n,args.k);
 
   if ((args.m == 0) || (args.n == 0)) return;
 
+#if 1
+#ifndef GEMM3M
+  if (args.m == 1) {
+	  char *NT=(char*)malloc(2*sizeof(char));
+	  if (transb&1)strcpy(NT,"T");
+	  else NT="N"; 
+//	  fprintf(stderr,"G E M V\n");
+	  GEMV(NT, &args.n ,&args.k, args.alpha, args.b, &args.ldb, args.a, &args.m, args.beta, args.c, &args.m);
+//SUBROUTINE SGEMV(TRANS,M,N,ALPHA,A,LDA,X,INCX,BETA,Y,INCY)
+return;
+  } else {
+  if (args.n == 1) {
+#ifndef CBLAS
+	  char *NT=(char*)malloc(2*sizeof(char));
+	  strcpy(NT,"N"); 
+#else
+	  char *NT=(char*)malloc(2*sizeof(char));
+	  if (transb&1)strcpy(NT,"T");
+	  else strcpy(NT,"N"); 
+#endif
+//	  fprintf(stderr,"G E M V ! ! ! lda=%d ldb=%d ldc=%d\n",args.lda,args.ldb,args.ldc);
+	  GEMV(NT, &args.m ,&args.k, args.alpha, args.a, &args.lda, args.b, &args.n, args.beta, args.c, &args.n);
+//SUBROUTINE SGEMV(TRANS,M,N,ALPHA,A,LDA,X,INCX,BETA,Y,INCY)
```

Comment:
I was thinking about this a little more.  For cases when n = 1, seems like this should be pretty straightforward, and parameters should just directly translate from GEMM to GEMV.

For cases when m = 1 I think there is a challenge.  in gemm (C:=AB) A is the vector, but GEMV (y:=Bx). so I think we need to re-arrange the equation by transposing (or, un-transposing) B.

Reply:
As a general word of caution, be careful with introducing any extraneous transposes for gemm variants dealing with complex numbers (`cgemm` and `zgemm`).  

For complex vectors and matrices, the conjugate transpose is typically used, where the sign of the imaginary components is flipped.  As an example, treating a column vector as a row vector (and vice versa) for complex numbers may not be simply a matter of interpreting the dimensions differently.   
https://en.wikipedia.org/wiki/Conjugate_transpose

To reduce risk, I'd suggest to completely leave out `cgemm` and `zgemm` from the proposed optimisation path (ie. forwarding gemm calls to gemv).

Reply:
@conradsnicta right, likely the performance gains would be much smaller for the complex cases anyway do to the extra coding necessary. So far I haven't even found time to get back to this issue.